File: src/binarytree/lc_1325_delete_leaves_with_given_value/delete_leaves_with_given_value.py
from treeNode import TreeNode

class Solution:
    def removeLeafNodes(self, root: TreeNode, target: int) -> TreeNode:
        
        def dfs(node):
            if not node:
                return None

            if node.val == target and not node.left and not node.right:
                return None
            
            root = TreeNode(node.val)
            root.left = dfs(node.left)
            root.right = dfs(node.right)
            if not root.right and not root.left and root.val == target:
                return None
            return root
        return dfs(root)

File: src/binarytree/lc_1325_delete_leaves_with_given_value/delete_leaves_with_given_value.go
type TreeNode struct {
	Val int
	Left *TreeNode
	Right *TreeNode
}
	
func removeLeafNodes(root *TreeNode, target int) *TreeNode {
	if root == nil {
		return nil
	}
	if root.Val == target && root.Left == nil && root.Right == nil {
		return nil
	}
	node := new(TreeNode)
	node.Val = root.Val
	node.Left = removeLeafNodes(root.Left, target)
	node.Right = removeLeafNodes(root.Right, target)
	if node.Left == nil && node.Right == nil && node.Val == target {
		return nil
	}
	return node
}